source/postprocess/visualization/spherical_velocity_components.cc

```diff
@@ -0,0 +1,125 @@
+/*
+  Copyright (C) 2011 - 2019 by the authors of the ASPECT code.
+
+  This file is part of ASPECT.
+
+  ASPECT is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2, or (at your option)
+  any later version.
+
+  ASPECT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file LICENSE.  If not see
+  <http://www.gnu.org/licenses/>.
+*/
+
+
+#include <aspect/postprocess/visualization/spherical_velocity_components.h>
+#include <aspect/utilities.h>
+
+
+namespace aspect
+{
+  namespace Postprocess
+  {
+    namespace VisualizationPostprocessors
+    {
+      template <int dim>
+      SphericalVelocityComponents<dim>::
+      SphericalVelocityComponents ()
+        :
+        DataPostprocessorVector<dim> ("spherical_velocity_components",
+                                      update_quadrature_points)
+      {}
+
+
+
+      template <int dim>
+      void
+      SphericalVelocityComponents<dim>::
+      evaluate_vector_field(const DataPostprocessorInputs::Vector<dim> &input_data,
+                            std::vector<Vector<double> > &computed_quantities) const
+      {
+        const unsigned int n_quadrature_points = input_data.evaluation_points.size();
+        Assert (computed_quantities.size() == n_quadrature_points,    ExcInternalError());
+        Assert (computed_quantities[0].size() == dim,    ExcInternalError());
+
+        Tensor<1,dim> velocity;
+        for (unsigned int q=0; q<n_quadrature_points; ++q)
+          {
+            for (unsigned int d = 0; d < dim; ++d)
+              {
+                velocity[d] = input_data.solution_values[q][this->introspection().component_indices.velocities[d]];
+
```

Comment:
I think you need to close the inner-most for loop here.

Reply:
hopefully got it right :)

Reply:
You know that you are forcing me to ask you to write a unit test now, right? :-)

Reply:
I need to write a test for another pull request. Can't you merge this and we'll get started on the tests ? :)